knees4bees/overlook
# Feature/iteration2

## What does this PR do? ##
- Gets close to completing iteration 2 (still need to attach event listener to rooms container and POST bookings data, need to add a couple media queries, and need to fix weird accessibility issue).
- Adds `RoomsRepo.js` and associated test; also moves a few functions to the `helpers.js` file.
- Adds display of available rooms for a selected date with ability to filter based on room types.
- Makes styling updates for improved accessibility and responsiveness.

## How should it be tested? ##
1. Fetch and install my code:
  - `cd` into the directory for my project
  - `git fetch`
  - `git checkout feature/iteration2`
  - in a second Terminal tab, `npm start` (you may need to `ctrl c` to stop your own local server from running your version of the project)
2. Confirm that all tests are passing when you run `npm test`.
3. In a browser, go to `http://localhost:8080/` and confirm that the needed data is being displayed. Try searching for rooms on different days, and then try filtering by different room types. Confirm that if no room types are selected an apology message is displayed (it may not be fierce enough though).
4. Take a look at the class file `RoomsRepo.js` and see if you spot anything that looks like a bad move or a code smell.

I'm getting a weird error in my Lighthouse accessibility audit where it says "form elements do not have associated labels". I added labels to both form elements on the landing page, and I'm still getting this error. I checked that each element has an `id` that matches its associated label's `for` attribute. Thoughts?
